### PR TITLE
add bootstrap docs and make git-repository mandatory

### PR DESCRIPTION
**Pull request by aclevername**
### Description

Feature PR: https://github.com/weaveworks/pctl/pull/244

I'm 99% sure that the `--git-repository` flag is not always required as part of the changes for `dependsOn`

### Checklist
- [ ] Added tests that cover your change
- [ ] Added/modified documentation as required (such as the `README.md` (diagrams, usage, roadmap etc), and anything in `examples/`)
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] Added a `kind` label to the PR (e.g. `kind/feature`)

### CODE REVIEW

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (nikimanoledaki) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:31]: A bit more info about how this is actually referring to flux namespace would be great!
Also what is `<name>` referring to?

Comment (nikimanoledaki) [reply userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:31]: Also is `<path-to-git-repo>` a relative path? So if you were to use this command in the root of your repo it would be `.`? On second thought, maybe this would be clearer if it were a real-life example with some commented out explanation 🤔

Comment (Callisto13) [reply userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md]: > A bit more info about how this is actually referring to flux namespace would be great!
Also what is <name> referring to?

I think this information exists somewhere else on the page? Maybe copy or rearrange?

Comment (aclevername) [reply userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md]: I did point to it, but I've also provided a brief summary again

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Callisto13) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:122]:
```suggestion
To avoid setting this flag on every call of `pctl add`, users can bootstrap the repository; see [here](#bootstrapping-your-local-git-repository).
```

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Callisto13) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:121]:
```suggestion
This flag is required for installing profiles because pctl generates flux `Kustomization` resources for
deploying the profile artifacts, and these resources need to know which `GitRepository` resource is governing the repository.
```

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Callisto13) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:28]:
```suggestion
you have bootstrapped your local git repository. The `--git-repository` references the namespace and name of the Flux `GitRepository`
```

**Comment by Callisto13** [comment]
Quick Q:

Is the config file generated here required by subsequent commands? I mean: if I don't bootstrap will later things fail? Can I still use the git-repo flag with `add`?

I think the answer to this is "yes", based on these changes, so my follow up would be: why would we not require that bootstrap always be run?

**Comment by aclevername** [comment]
> Quick Q:
> 
> Is the config file generated here required by subsequent commands? I mean: if I don't bootstrap will later things fail? Can I still use the git-repo flag with `add`?
> 
> I think the answer to this is "yes", based on these changes, so my follow up would be: why would we not require that bootstrap always be run?

if `--git-repository` is provided it takes precedence, otherwise we look for the configfile. https://github.com/aclevername/pctl/blob/602fb9d32848c767a8b51a7ae50a9b41fca757db/cmd/pctl/add.go#L177-L194

**Comment by aclevername** [comment]
> > Quick Q:
> > Is the config file generated here required by subsequent commands? I mean: if I don't bootstrap will later things fail? Can I still use the git-repo flag with `add`?


if `--git-repository` is provided it takes precedence, otherwise we look for the configfile. https://github.com/aclevername/pctl/blob/602fb9d32848c767a8b51a7ae50a9b41fca757db/cmd/pctl/add.go#L177-L194

> > I think the answer to this is "yes", based on these changes, so my follow up would be: why would we not require that bootstrap always be run?

:shrug: I just think for quick one-off `pctl add <blah>` its nice to not have to do this. Practically I think everyone will always bootstrap

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Callisto13) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:28]: This opener is confusing, perhaps:
```suggestion
You can use the `flux bootstrap` command to save commonly used `pctl` configuration to your GitOps repo.

Once such piece of configuration is your repo's `GitRepository` reference.
When installing profiles you will have to provide the `--git-repository` flag (See [below](#the-git-repository-flag)) unless
you have bootstrapped your local git repository. The `--git-repository` references the namespace and name of the
```

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Callisto13) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:40]: That last bit is obvious I think
```suggestion
`pctl add` commands in this repository will then detect the pctl config and re-use the value.
```

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Callisto13) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:37]:
```suggestion

_See [below](#the-git-repository-flag) for how to find the namespace and name of your `GitRepository` resource._

```

or something.

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Callisto13) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:31]:
```suggestion
If you have **not** bootstrapped your local GitOps repository, you will have to provide the `--git-repository` flag when installing profiles (see [below](#the-git-repository-flag) for more detail). The `--git-repository` references the namespace and name of the
```

**Review by Callisto13 — APPROVED** [review]

**Comment by aclevername** [comment]
merging blocked on https://github.com/weaveworks/pctl/pull/244

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Skarlso) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:27]: Shouldn't that be `pctl bootstrap`... ?

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Skarlso) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:29]:
```suggestion
Once such piece of configuration is your flux repo's `GitRepository` resource.
```

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Skarlso) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:37]: This deviates from `add` as in, it accepts an absolute path instead of a relative one. Not a problem, just wanted to say that. :D

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (aclevername) [reply userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:27]: lmao whoops

Comment (Callisto13) [reply userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:27]: see, this is what you get when you don't read my suggestions before merging. i hope you have learned your lesson.

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Callisto13) [reply userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:29]: Maybe `GitOps` repo to be consistent. we have not called it Flux repo anywhere else

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (Skarlso) [inline userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:126]: Should we add something like, avoid setting this flag FOR A REPOSITORY?

Because this feels like it's suggesting that you won't have to ever set it ever again. You know what I mean? That this is a local setting not a global one.

**Review by Skarlso — APPROVED** [review]
Some comments and questions otherwise it looks okay. :)

File: userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md
Comment (aclevername) [reply userdocs/profiles.dev/docs/installer-docs/installing-via-gitops.md:37]: It accepts both I believe, I just gave absolute as an example. `.` is also valid

**Pull request merged**
